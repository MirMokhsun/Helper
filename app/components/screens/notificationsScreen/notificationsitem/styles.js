import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'column',
        width: '100%',
        height: 100,
        paddingLeft: 10,
        paddingRight: 10,
        marginTop: 5,
    },
    dataContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    containerBackgroundLight: {
        backgroundColor: '#ffffff',
        borderBottomColor: '#D3D9E7',
        borderBottomWidth: 1,
    },
    textDate: {
        color: "#1459D2",
        fontSize: 11,
        lineHeight: 13,
        fontFamily: 'Roboto',
        marginTop: 10
    },
    circleContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'flex-end',
    },
    circle: {
        backgroundColor: '#F86F6F',
        borderRadius: 50,
        width: 8,
        height: 8,
    },
    textDescription: {
        color: "#000756",
        fontSize: 16,
        lineHeight: 19,
        fontFamily: 'Roboto',
        marginTop: 5,
    },
});