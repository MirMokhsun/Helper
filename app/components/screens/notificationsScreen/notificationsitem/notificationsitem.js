import React, { PureComponent } from 'react';
import { View, Text, TouchableWithoutFeedback } from 'react-native';
import { styles } from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withNavigation } from 'react-navigation';
import NotificationApi from '../../../../api/notification.api/notification.api';
import { addPushNotification } from '../../../../api/notification.api/redux/addPushNotificationAction';


export class NotificationsItem extends PureComponent {
    constructor(props) {
        super(props)
        this.notificationApi = new NotificationApi();
    }

    openNotificationsItemScreen = () => {
        const { navigation, addPushNotification, notification } = this.props;
        addPushNotification(notification);
        navigation.navigate('NotificationsItemDetailsScreen', notification);
    };

    render() {
        const { textDate, title, wasRead } = this.props;
        return (
            <TouchableWithoutFeedback onPress={this.openNotificationsItemScreen} testID='NotificationsItemDetailsScreenID' accessibilityLabel='NotificationsItemDetailsScreenID'>
                <View style={[styles.container, styles.containerBackgroundLight]}>
                    <View style={styles.dataContainer}>
                        <View >
                            <Text style={styles.textDate}>{textDate}</Text>
                        </View>
                        <View style={styles.circleContainer} >
                            {wasRead ? <View /> : <View style={styles.circle} />}
                        </View>
                    </View>
                    <View>
                        <Text style={styles.textDescription} numberOfLines={3} >{title}</Text>
                    </View>
                </View>
            </TouchableWithoutFeedback>
        );
    }
};

const mapDispatchToProps = dispatch => bindActionCreators({
    addPushNotification,
}, dispatch);

export default withNavigation(connect(null, mapDispatchToProps)(NotificationsItem));