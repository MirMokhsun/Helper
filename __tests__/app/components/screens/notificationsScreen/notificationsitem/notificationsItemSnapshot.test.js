import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { NotificationsItem } from '../../../../../../app/components/screens/notificationsScreen/notificationsitem/notificationsitem';
jest.mock('react-navigation');
it('create snapshot NotificationsItem', () => {
    let  textDate = "Feb 25, 2018", body = "Hello" , isRead =true;
   let  notification = {
        MsgText:"Hello"
    };
    const renderer = new ShallowRenderer()
    let notificationsItem = renderer.render(<NotificationsItem notification={notification} textDate ={ textDate } MsgText ={body} isRead = { isRead }/>);
    expect(notificationsItem).toMatchSnapshot();
});