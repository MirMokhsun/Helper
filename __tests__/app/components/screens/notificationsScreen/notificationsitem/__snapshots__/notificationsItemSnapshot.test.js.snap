// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot NotificationsItem 1`] = `
<TouchableWithoutFeedback
  accessibilityLabel="NotificationsItemDetailsScreenID"
  onPress={[Function]}
  testID="NotificationsItemDetailsScreenID"
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "height": 100,
          "marginTop": 5,
          "paddingLeft": 10,
          "paddingRight": 10,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#D3D9E7",
          "borderBottomWidth": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#1459D2",
              "fontFamily": "Roboto",
              "fontSize": 11,
              "lineHeight": 13,
              "marginTop": 10,
            }
          }
        >
          Feb 25, 2018
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F86F6F",
              "borderRadius": 50,
              "height": 8,
              "width": 8,
            }
          }
        />
      </View>
    </View>
    <View>
      <Text
        numberOfLines={3}
        style={
          Object {
            "color": "#000756",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "lineHeight": 19,
            "marginTop": 5,
          }
        }
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;
